### ISSUE TITLE
Bitbucket Support

### ISSUE THREAD

**Issue by silverskyvicto** [earthly/earthly#482]
Do you support integration with bitbucket?

**Comment by alexcb** [comment]
Hi @silverskyvicto , not currently, but that's something we can add. To better understand the use-case, is your bitbucket git repo public or private? are you using bitbucket.org, or a self-hosted version?

**Comment by silverskyvicto** [comment]
Hi @alexcb 
Thanks for the comment.

I use both `bitbucket.org` and self-hosted, but I'd appreciate it if `bitbucket.org` was supported first.

Git repository is private.

**Comment by alexcb** [comment]
Ideally we should be able to support any self-hosted repos, and then we'll have a pre-populated table of mappings for github, gitlab, bitbucket etc. I'll add this to our TODO list, and see what we can come up with. Thanks for the request.

**Comment by alexcb** [comment]
We're currently waiting on some changes to buildkit in https://github.com/moby/buildkit/pull/1782 once that PR is in place we'll be able to work towards supporting this feature.

**Comment by alexcb** [comment]
We've been busy behind the scenes doing some yak-shaving, and wrote a new integration test for building against private repos; now that that's in main, we have a new PR that works towards incorporating the buildkit changes: https://github.com/earthly/earthly/pull/518

**Comment by alexcb** [comment]
We just merged a new experimental setting for custom or private repos in our main branch under https://github.com/earthly/earthly/pull/611 It has yet to be documented, but the PR gives an example of how to configure git repos that aren't supported out of the box.
